File: src/main/scala/com/example/rest/MyUser.java
package com.example.rest;

import java.security.Principal;
import java.util.Random;

public class MyUser implements Principal {
	private final String username;
	private final int weight;
	private static final Random rnd = new Random();
	
	public MyUser(String username) {
		this.username = username;
		this.weight = 40 + 60 * rnd.nextInt();
	}
	
	public String getUsername() {
		return username;
	}
	
	public int getWeight() {
		return weight;
	}

	@Override
	public String getName() {
		return username;
	}
}

File: src/main/scala/com/example/rest/MyUser.scala
package com.example.rest;

import java.security.Principal;
import java.util.Random;

object MyUser {
  private val rnd = new Random()
}

class MyUser(val username: String) extends Principal {
  val weight = 40 + 60 * MyUser.rnd.nextInt()

  override def getName() = username
}
